File: lib/ts/src/internal/generation/GenerateRandomBytes.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { GenerateContext } from '../../internal/generation/GenerateContext.js';

export function generateRandomBytes(blueprintValue: any, useBlueprintValue: boolean, ctx: GenerateContext): any {
    if (useBlueprintValue) {
        return blueprintValue;
    } else {
        return ctx.randomGenerator.nextBytes();
    }
}

File: lib/go/internal/types/GenerateRandomBytes.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// GenerateRandomBytes returns a pseudo-random byte slice, optionally reusing a blueprint value.
func GenerateRandomBytes(blueprintValue any, useBlueprintValue bool, ctx *GenerateContext) any {
	if useBlueprintValue {
		return blueprintValue
	}
	if ctx == nil || ctx.RandomGenerator == nil {
		return nil
	}
	return ctx.RandomGenerator.NextBytes()
}
